### ISSUE TITLE
Data corrupts after leaving friends town or  when a random error happens on the island.

### ISSUE THREAD

**Issue by Derek1199** [RyDog199/ACNL-NTR-Cheats#36]
The data corrupts after leaving a friend's town usually and is instantly fixed once I reboot my 3ds. According to the issue game data can no longer be read once this happens. I'm using the American version since I have a US 3ds. I'm using the latest commit btw. Anything you can do?

**Comment by RyDog199** [comment]
Do you connect with the debugger/ntrclient? 
If not, please enable exception handlers/errordisp in Luma settings and upload a crash dump after the next crash so I can further investigate this issue. 
You're the first person to have this issue, so it could also be a faulty SD card, faulty 3DS cart (if you're using one) or something is wrong with your 3DS. 
I won't be able to investigate this issue until more information is provided (ie: the luma dumps I mentioned before)

**Comment by Derek1199** [comment]
Its not a crash. It's just corrupting, but instantly fixed on a reboot.
Maybe from options I'm enabling inside the cheats menu and/or from using
the latest commit

**Comment by Derek1199** [comment]
Well, i'll compile this myself and see if I still get the error. I know how to compile stuff like armips and Luma, so compiling this should be easy.

**Comment by Derek1199** [comment]
This issue is still occuring when leaving a friend's town, but not the island anymore in the latest release. However, now there is another issue in the latest release. Using one of the money codes triggers a crash. I'll upload the crash dump of the money crash, then after I'll upload the crash dump. I parsed the crash dump myself, but cannot understand the debug output. So, i'll send it to you here.

**Comment by Derek1199** [comment]
[crash_dump_00000207.zip](https://github.com/RyDog199/ACNL-NTR-Cheats/files/1220246/crash_dump_00000207.zip)

**Comment by Slattz** [comment]
@Derek1199 Does the issue still occur on the latest versions of the plugin?

**Comment by kokichioumalovesme** [comment]
I dont know if someone has already said this but the corruption is a false positive. I get this too (even on the newest plug-in) and I just reboot my 3DS, reboot NTR, reboot ACNL, and all my data is still there. It’s a pain to reboot every time but it’s reassuring to know that nothing actually corrupted

**Comment by Derek1199** [comment]
Yeah this describes everything that happened to me. I don't play anymore but what was above my comment just now confirms it still happens. The process now is to find out what causes it.

**Comment by LeviSpecialzz** [comment]
My data doesn't get corrupted, but it used to sometimes. If you haven't already, I recommend using something like JKSM or Checkpoint to restore your save data when it gets corrupted.

**Comment by Derek1199** [comment]
i already did that before. But even when i did that when I did play it
still would always happen. Although after rebooting and starting again
everything would appear as normal. But the fact still stands that after an
error happens while running the cheats in a friend's town, the data comes
off as unreadable and it asks you if you wanna delete the data. So this
should be figured out in some way if possible as to what causes it

**Comment by LeviSpecialzz** [comment]
What cheats are you running in your friend's town?

**Comment by Derek1199** [comment]
it was too long ago since i played. But i just used the ordinary most
common cheats that were used. Nothing really excessive. Just having the
plugin enabled seemed to do it. I can't test now unfortunately though. I
haven't played my 3ds in over 6 months. I pretty much quit it forever. But
i had the most common used cheats enabled.

**Comment by LeviSpecialzz** [comment]
At the time of making the original post, you had version 3.2.1, correct?

**Comment by Derek1199** [comment]
i believe so, yeah

**Comment by realMBite** [comment]
Have you updated since using that version? If so, what version do you use now?

**Comment by Derek1199** [comment]
Last version I used was the beta 2 or 3 lol. From inside the acnl modding
server. And that issue still did happen. I haven't played my 3ds in over 8
months though. But yeah it was still present even back then

**Comment by realMBite** [comment]
Well, if you still care about this, I recommend switching to 4.1.1.

**Comment by Derek1199** [comment]
Yeah I know. I would if I still played to test it. But I don't play
anymore. Lost motivation to play in the end. But if others with the same
issue can test it then that's good and fine.

**Comment by realMBite** [comment]
Okay, if you don't play this anymore, then you can close this issue and move on. You still play your 3DS in general or completely stopped?

**Comment by Derek1199** [comment]
Yeah I don't play my 3ds in general anymore. But someone else here did say
they had the same issue. Should leave this open in case they can test it
for now.

**Comment by realMBite** [comment]
Okay. best to you.

**Comment by kokichioumalovesme** [comment]
I am using 4.1.1 and this still happens to me. I have not tested about if the corruption happens without NTR booted up or not. I have enabled Catalog To Pockets, sometimes Time TrAvel but not always, Text To Item, all four of the infinite currency settings, inventory box extender, sometimes Real Time World edit, sometimes Special NPC Changer, sometimes access catalog and storage anywhere, faint, pick every tour, weather modifier, keyboard extender, sometimes amiibo spoofer, and touch coordinates. However, even when I have everything disabled but the plug-in is loaded, it still happens, so I don’t believe any of the cheats themselves are causing issues
